src/App.tsx

```diff
@@ -1,24 +1,329 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useEffect,
+  useState,
+  useCallback,
+  // useMemo,
+} from 'react';
+import classNames from 'classnames';
+import './App.scss';
+
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
+import { BASE_URL } from './utils/fetchClient';
 
-const USER_ID = 0;
+import { ErrorType } from './types/ErrorType';
+import { ErrorShow } from './components/ErrorShow';
+
+import { TodoList, USER_ID } from './components/TodoList/TodoList';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [addingTodoTitle, setAddingTodoTitle] = useState('');
+  const [errorToShow, setErrorToShow] = useState<ErrorType>('none');
+  const [todosToShow, setTodosToShow] = useState(todos);
+  const [chosenFilter, setChosenFilter] = useState<string>('all');
+  const [creating, setCreating] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processings, setProcessing] = useState<number[]>([]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((todosFromServer: Todo[]) => setTodos([...todosFromServer]));
+  }, [todos]);
+
+  useEffect(() => {
+    const filteredTodos = todos.filter(todo => {
+      switch (chosenFilter) {
+        case 'active':
+          return !todo.completed;
+        case 'completed':
+          return todo.completed;
+        case 'all':
+        default:
+          return todo;
+      }
+    });
+
+    setTodosToShow(filteredTodos);
+  }, [todos, chosenFilter]);
+
+  const handleSetNewTitle = useCallback((
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const { target: { value } } = event;
+
+    return setAddingTodoTitle(value);
+  }, []);
+
+  const getNewTodoId = useCallback(() => {
+    return Math.max(...todos.map(todo => todo.id)) + 1;
+  }, []);
```

Comment:
```suggestion
  const getNewTodoId = useCallback(() => {
    return Math.max(...todos.map(todo => todo.id)) + 1;
  }, [todos]);
```

Reply:
better move this function to helpers

Reply:
done
and even with JSDocs comments
to get more familiar with this toolkit and to make my knowledge scope wider
![image](https://user-images.githubusercontent.com/120763734/232980207-72670d10-ef4d-486f-9723-68616a4cad5c.png)